community-tutorials/web/01-en.md

```diff
@@ -0,0 +1,404 @@
+a2dismod php*
+a2enmod php8.0
+service apache2 restart
+service php8.0-fpm restart
+curl -sS https://getcomposer.org/installer | sudo php -- --install-dir=/usr/local/bin --filename=composer
+{ crontab -l 2>/dev/null; echo "$(( $RANDOM % 60 )) $(( $RANDOM % 3 + 3 )) * * * curl -sS https://getcomposer.org/installer | sudo php -- --install-dir=/usr/local/bin --filename=composer" ; } | crontab -
+```
+
+## Step 1.4 Installing MariaDB
+```sh
+apt install mariadb-server mariadb-client -y
+```
+
+## Step 1.4.1 Securing MariaDB
+```sh
+mysql_secure_installation
+```
+Press Enter <br>
+Press y <br>
+Press y <br>
+Set a Password <br>
+Press y <br>
+Press y <br>
+Press y <br>
+Press y <br>
+## Step 1.4.2 Create a user with root permissions on MariaDB
+```sh
+mysql -u root -p
+```
+Enter the password from step 1.4.1 <br>
+Run the following command and replace "USERNAME" with an username do not use `root` as username and replace "PASSWORD" with a secure password
+```sh
+CREATE USER 'USERNAME'@'localhost' IDENTIFIED BY 'PASSWORD';
+GRANT ALL PRIVILEGES ON *.* TO 'Username'@'localhost' WITH GRANT OPTION;
+FLUSH PRIVILEGES;
+exit;
+```
+
+## Step 1.5 Installing a configuration for the Webserver 
+First make sure your domain or subdomain is located to your server in its DNS settings <br>
+
+Replace every `DOMAIN` with your domain or subdomain in the next commands <br/>
+Generate a SSL-Certificate with `certbot certonly --apache -d DOMAIN` <br/>
+Run `nano /etc/apache2/sites-enabled/DOMAIN.conf` <br/>
+Paste now this in the file:
+```
+<VirtualHost *:80>
+    ServerName DOMAIN
+    Redirect permanent / https://DOMAIN/
+	Protocols h2 h2c http/1.1
+    ErrorLog /var/log/apache2/DOMAIN.error.log
+</VirtualHost>
+
+<VirtualHost *:443>
+    ServerName DOMAIN
+    DocumentRoot /var/www/FOLDER
+	Protocols h2 h2c http/1.1
+    ErrorLog /var/log/apache2/DOMAIN.error.log
+SSLCertificateFile /etc/letsencrypt/live/DOMAIN/fullchain.pem
+SSLCertificateKeyFile /etc/letsencrypt/live/DOMAIN/privkey.pem
+</VirtualHost>
+```
+Replace every `DOMAIN` with your domain and set a folder path under "DocumentRoot", all files in this folder will be available on your DOMAIN in the web <br/>
+Save your edit with ctrl + x, then press y and now save it with enter <br/>
+Now create this folder with `mkdir /var/www/FOLDER` <br>
+Now restart Apache2 with `service apache2 restart` <br>
+
+If you are doing this direct on a domain and NOT on a subdomain, I recommend to repeat this step with www.DOMAIN
+
+## Step 1.6 - Installing PHPMyAdmin - Optional
+```sh
+curl -L https://www.phpmyadmin.net/downloads/phpMyAdmin-latest-all-languages.zip -o phpmyadmin.zip
+unzip phpmyadmin.zip
+rm phpmyadmin.zip
+mv phpMyAdmin-*-all-languages pma
+mv pma /var/www
+nano /var/www/pma/config.inc.php
+```
+Now add here the following lines and change `x1rfsRmlidz69R3vKvxpDVl6lbOReX0Emc3sVD0ZkpXh4ek12zMZCWJJi722Se36M0fiRdRDPk05ePcO4W0l8bP8vJMcMhoLVSs` to make it unique
```

Comment:
What are we doing here?

Reply:
this is to make the PHPMyAdmin cookies more secure